### ISSUE TITLE
[Bug]: AUROC miscalculation when using small validation batchsize

### ISSUE THREAD

**Issue by brein0453** [openvinotoolkit/anomalib#2003]
### Describe the bug

I'm using anomalib v 1.0.1, and found AUROC value miscalculation when validation batch size is small.

I think torchmetric AUROC causes this error
https://lightning.ai/docs/torchmetrics/stable/classification/auroc.html

### Dataset

Other (please specify in the text field below)

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

this is simple example.
batchsize=1 and scores in [-2, 1.8]

```python
>>> from anomalib.metrics import AUROC
>>> import torch
>>> auroc = AUROC()
>>> for pred in zip(torch.arange(-2, 2, 0.2)):
...     target = pred > 0
...     auroc.update(pred, target)
>>> auroc.compute()
tensor(0.8990)


>>> from sklearn.metrics import roc_auc_score
>>> preds = []
>>> targets = []
>>> for pred in torch.arange(-2, 2, 0.2):
...     preds.append(pred)
...     targets.append(pred > 0)
>>> roc_auc_score(targets, preds)
>>> 1.0
```

### OS information

.

### Expected behavior

.

### Pip/GitHub

pip

### What version/branch did you use?

1.0.1

### Configuration YAML

```yaml
.
```

### Logs

```shell
.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
This seems quite problematic, but I am not sure what would be the reason for this behavior. Anomalib AUROC doesn't use torchmetrics AUROC, but rather BinaryROC and auc combo [source](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/metrics/auroc.py).

**Comment by blaz-r** [comment]
But sklearn function does that as well, doesn't it?

**Comment by brein0453** [comment]
sklearn.metrics.roc_auc_score does not apply sigmoid at preds.

I fixed this by appling sigmoid at my model's score. I think it would be better to deal with this issue in torchmetric. Thank you for your response.

**Comment by blaz-r** [comment]
I actually tested your code and after some debugging I found the reason. Problem here is that if you update values one by one, the sigmoid is only applied if value is <0 or >1. This assumption is broken in your example when the range is from 0 to 1, so the sigmoid is not used at all. (for example. 0.1 stays 0.1, but with sigmoid that is 0.525). This is a bit of a weird case, but I think it's the expected behavior. To get the same results, you can apply the same storing procedure as in case of sklearn:

```python
from anomalib.metrics import AUROC
import torch
auroc = AUROC()
preds = []
targets = []
for pred in torch.arange(-2, 2, 0.21):
    preds.append(pred)
    targets.append(int(pred > 0))

auroc.update(torch.tensor(preds), torch.tensor(targets))
auroc.compute()
```

This now gives the same result, as the check for all >= 0 and <= 1 fails so sigmoid is applied to all elements in this case, resulting in the desired behavior. 
Another correct solution for small batches would thus be a manual usage of sigmoid before the update call.
